sys/shell/shell.c

```diff
@@ -45,6 +46,15 @@ static void _putchar(int c) {
 #endif
 #endif
 
+/* on native, stop RIOT on EOF */
+#ifndef SHELL_SHUTDOWN_ON_EOF
+#ifdef CPU_NATIVE
+#define SHELL_SHUTDOWN_ON_EOF   (1)
+#else
+#define SHELL_SHUTDOWN_ON_EOF   (0)
+#endif
```

Comment:
```suggestion
#define SHELL_SHUTDOWN_ON_EOF
```

Reply:
Ok, removed the overwritability.

Reply:
Hm that was not what I meant. The suggestion was to do:

```
#ifndef SHELL_SHUTDOWN_ON_EOF
#  ifdef CPU_NATIVE
#    define SHELL_SHUTDOWN_ON_EOF
#  endif
#endif
```

as it is equivalent (how it is used here), but shorter and more ideomatic than how it was before:

```
#ifndef SHELL_SHUTDOWN_ON_EOF
#  ifdef CPU_NATIVE
#    define SHELL_SHUTDOWN_ON_EOF (1)
#  else
#   define SHELL_SHUTDOWN_ON_EOF (0)
#  endif
#endif
```

The overridability is important, but there's no need to define to `(1)`, and no need for an else case to define to `(0)`. We've had that before, no? If you disagree, please open an issue, but until that is resolved, we stick to `#define FOO` instead of `#define FOO 1`.

BTW, wasn't the check below nonfunctional in the negative case (always positive) as it was before? `#ifdef SHELL_SHUTDOWN_ON_EOF` evaluates to 1 if `SHELL_SHUTDOWN_ON_EOF` is defined to `(0)`.